src/View/ViewableData.php

```diff
@@ -66,6 +66,11 @@ class ViewableData implements IteratorAggregate
      */
     private static $casting_cache = [];
 
+    /**
+     * Acts as a PHP 8.2+ compliant replacement for dynamic properties
+     */
+    private array $data = [];
```

Comment:
ViewableData is a base class and there's simply far too many magic calls in Silverstripe to full stop using dynamic properties.  Realistically it's either a `private array` or just add `#[\AllowDynamicProperties]` to the class